app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt

```diff
@@ -168,7 +179,22 @@ class NewProfileFragment : Fragment() {
                 }
             }
         }
-
+        val args = arguments?.getParcelable<Profile>("Profile")
+        val daysSelected = Gson()
+        val type = object : TypeToken<List<Boolean>>() {}.type
```

Comment:
Declare daysSelected and type outside the onCreateView() and lazy initialize them.